src/citrine/resources/dataset.py

```diff
@@ -221,6 +221,16 @@ def register(self, data_concepts_resource: ResourceType) -> ResourceType:
         if isinstance(data_concepts_resource, ConditionTemplate):
             return self.condition_templates.register(data_concepts_resource)
 
+    def register_all(self, data_concepts_resources: List[ResourceType]) -> List[ResourceType]:
+        """
+        Register multiple data concepts resources to each of their appropriate collections.
+
+        :param data_concepts_resources: the resources to register. Can be different types.
+
+        :return the registered versions
+        """
+        return [self.register(resource) for resource in data_concepts_resources]
```

Comment:
I believe you will run into a problem here: For example, consider a MaterialSpec linked to a ProcessSpec. Both of the items are in the `data_concepts_resources` list, but you will get an error if MaterialSpec is registered before ProcessSpec.

In devkit, I filtered by each type, then registered in an order so I don't get an error. (See [this function](https://github.com/CitrineInformatics/nextgen-devkit/blob/master/seeding/register.py#L19))